### ISSUE TITLE
[Question] Unable to get button click on analytics report. [Help Wanted] 

### ISSUE THREAD

**Issue by wadezone** [microsoft/playwright#3886]
Is there a community forum for playwright or is the GitHub the place to get help? 

I have been asked to figure out report automation testing for our analtyic reports on our internal company site. 
The reports are done in one of the major analytical tools available. 
I need to be able to search for something, drill through, get a value on the report and check it against a database. 
I haven't been able to find where others are doing something like this. 
Due to company policy and securities, I do have to be careful of what I post in my questions and examples. 

I tried using the selenium IDE and recorder, Cypress, and Playwright. 
I am having the most success with Playwright but continue to hit some road blocks. 
After serveral weeks of reading documentation and trying different things I found the Playwright CLI. 
This actually has helped verify a few things but keep running into similar issues where Playwright is not able to recognize something on the page/frame. 

I have a search button and a clear button on the report. I can 't get Playwright to click on either one of the buttons.

I have tried from the "page" and from the "frame" objects the different possible methods/events with no luck. 
I am succesful in finding the frame with the following lines of code

    const frame = page.frames().find(frame => frame.name() === 'visual-sandbox');
    console.log(frame.name());

I tried what the CLI recorded and when I used what it returned it failed too. 

After typing in the search button the value to search, I tried to do a "enter" which should also trigger the search button. 
The recorder didn't even capture this key press.
I have tried so many things that I didn't keep a running list. 

Examples below, I tried doing the same from the "await frame ...."

	await page.click('//button[normalize-space(.)="Cear"]');
	await page.dispatchEvent('button#c-glyph.search-button', 'click');
	await page.click('search-button');
	await page.press('#sandbox-host > div', "Enter");
	
	
	const [response] = await Promise.all([
    page.waitForNavigation(), // The navigation promise resolves after navigation has finished
    page.click('#sandbox-host > div > button.c-glyph.search-button'),
    //]);
    
I got the furthest with the following line where it looks like it knew what I was asking.  
   const hrefElement = await frame.$('#sandbox-host > div > button.c-glyph.search-button');
   await hrefElement.click();
```
(node:31864) UnhandledPromiseRejectionWarning: TimeoutError: elementHandle.click: Timeout 30000ms exceeded.
=========================== logs ===========================
[api] attempting click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
[api]   element is visible, enabled and does not move
[api]   scrolling into view if needed
[api]   done scrolling
[api]   checking that element receives pointer events at (948.83,198.06)
[api]   element does not receive pointer events
[api] retrying click action
[api]   waiting for element to be visible, enabled and not moving
============================================================
Note: use DEBUG=pw:api environment variable and rerun to capture Playwright logs.
```
Looking at the Chrome Dev Tools inspection, this is what I get for different portions of the element for search button. 

		SearchButtonOutter
		"copy selector"
		#sandbox-host > div > button.c-glyph.search-button

		"copy JS Path"
		document.querySelector("#sandbox-host > div > button.c-glyph.search-button")

		"copy element":
		<button class="c-glyph search-button" name="search-button">
		                                          <span class="x-screen-reader">Search</span>
		                                        </button> 

		SearchButtonInner
		there isn't anything to copy for the element, but I can get this from the inspect styles section
		.visual-textFilter25A4896A83E0487089E2B90C9AE57C8A .text-filter-search button.search-button:before {
 
What the Playwright CLI returns on the elements are the following:
//button[normalize-space(.)='Search']    
//button[normalize-space(.)='Cear']

I could use some help on the right way to reference the click method for the buttons.  I have attached screen shots to show you what I am trying to click on. 

Should I only post in one of the Playwright repos or is it a different set of developers for the Playwright CLI repo. 

I hope this is an appropriate location to ask for help. I will most likely post here on 2 other topics if it is. 
![ClearButtonRecording](https://user-images.githubusercontent.com/71278576/93219104-9c3dad00-f730-11ea-8ebe-85dec326193c.JPG)
![SearchButtonInner](https://user-images.githubusercontent.com/71278576/93219107-9cd64380-f730-11ea-84a4-e320f0f7a3c1.JPG)
![SearchButtonOutter](https://user-images.githubusercontent.com/71278576/93219108-9cd64380-f730-11ea-8a1e-5f59918b2184.JPG)
![SearchButtonrecorder](https://user-images.githubusercontent.com/71278576/93219109-9d6eda00-f730-11ea-83f0-126eea4d560a.JPG)

**Comment by osmenia** [comment]
Try this: 

  const test = await page.evaluate(() =>
        document.querySelector("YOUR CSS SELECTOR").click()
      );

**Comment by wadezone** [comment]
I apologize for the delay. I had not verified my email address and when I tried, my email service was down all day.  Thank you so much for your reply. I thought I had tried something like that but it definitely wasn't with the combination of the evaluate and the queryselector. 

It didn't work for page, but both buttons worked when I used frame. 

const test = await frame.evaluate(() =>
document.querySelector("#sandbox-host > div > button.c-glyph.search-button").click()
);

I was ecstatic when I got this work. Thank you so much. I haven't been able to resolve a couple of my other issues using the similar object. If no luck I would love to ask for help here again. 

Thank you again. 
I am really liking Playwright even though I fell lost sometimes. This is all new to me.
